pytorch_lightning/accelerators/accelerator.py

```diff
@@ -12,14 +12,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from abc import abstractmethod
-from typing import Any, Dict, Optional, Union
+from typing import Any, Dict, Union
 
 import torch
-from torch.nn import Module
 
 import pytorch_lightning as pl
-from pytorch_lightning.plugins.precision import PrecisionPlugin
-from pytorch_lightning.plugins.training_type import TrainingTypePlugin
 
 
 class Accelerator:
```

Comment:
I understand this PR is to facilitate the refactor. What should the ideal Accelerator API look like? It looks like the accelerator is now not much more than the torch.device being used, vendor specific setup, and a way to get device stats (which is not a bad thing)

Reply:
@ananthsub @justusschock, I agree that current Accelerator is more like a Device class and it's a good thing to have :) And more device related check/logic could be moved into this Accelerator subclasses after this 3/n. Especially as @justusschock mentioned logic like has_tpu(). Or mis configs/error specific to one Accelerator like cpu `num_processes` 
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/connectors/accelerator_connector.py#L262-L264




>Make this the only source of truth for the used device type (e.g either replace the enum, or make it a property of this class) as I find it kind of confusing, that we check different things here and they could potentially diverge (although they shouldn't :)

Totally agree, we are planning on deprecating the trainer._device_type from and unify the check for device type to check accelerator class [Remove trainer._device_type in favor of check Accelerator class](https://github.com/PyTorchLightning/pytorch-lightning/issues/11002) and [Generalize internal checks for precision plugin type, training type, accelerator type](https://github.com/PyTorchLightning/pytorch-lightning/issues/10821) 
+1 make device_type a property of Accelerator class and deprecate Accelerator enum. (I think there will be circle import issue to solve before we can deprecate accelerator_enum)